app/controllers/api/v2/courses_controller.rb

```diff
@@ -56,6 +56,14 @@ def update
         enrichment.assign_attributes(update_params)
         enrichment.save
 
+        site_ids = params[:course][:sites_ids]
+        if site_ids.present?
+          new_sites = site_ids
```

Comment:
Would this work too?

```suggestion
          new_sites = @provider.sites.where(id: site_ids.map(&:to_i))
```

Reply:
If that does, you could to the `to_i` conversion on line 59, which lends itself to @defong's suggestion too.

Reply:
FWIW seems to work fine without the `.to_i` conversion.